hugegraph-common/src/main/java/org/apache/hugegraph/rest/AbstractRestClient.java

```diff
@@ -207,31 +263,62 @@ public RestResult post(String path, Object object) {
     }
 
     @Override
-    public RestResult post(String path, Object object,
-                           MultivaluedMap<String, Object> headers) {
+    public RestResult post(String path, Object object, RestHeaders headers) {
         return this.post(path, object, headers, null);
     }
 
     @Override
-    public RestResult post(String path, Object object,
-                           Map<String, Object> params) {
+    public RestResult post(String path, Object object, Map<String, Object> params) {
         return this.post(path, object, null, params);
     }
 
+    private Request.Builder getRequestBuilder(String path, String id, RestHeaders headers,
+                                              Map<String, Object> params) {
+        HttpUrl.Builder urlBuilder = HttpUrl.parse(this.baseUrl).newBuilder()
+                                            .addPathSegments(path);
+        if (id != null) {
+            urlBuilder.addPathSegment(id);
+        }
+
+        if (params != null) {
+            params.forEach((name, value) -> {
+                if (value == null) {
+                    return;
+                }
+
+                if (value instanceof Collection) {
+                    for (Object i : (Collection<?>) value) {
+                        urlBuilder.addQueryParameter(name, String.valueOf(i));
+                    }
+                } else {
+                    urlBuilder.addQueryParameter(name, String.valueOf(value));
+                }
+            });
+        }
+
+        Request.Builder builder = requestBuilder.url(urlBuilder.build());
+
+        if (headers != null) {
+            builder.headers(headers.toOkhttpHeader());
+        }
+
+        this.attachAuthToRequest(builder);
+
+        return builder;
+    }
+
+    @SneakyThrows
     @Override
-    public RestResult post(String path, Object object,
-                           MultivaluedMap<String, Object> headers,
+    public RestResult post(String path, Object object, RestHeaders headers,
                            Map<String, Object> params) {
-        Pair<Builder, Entity<?>> pair = this.buildRequest(path, null, object,
-                                                          headers, params);
-        Response response = this.request(() -> {
-            // pair.getLeft() is builder, pair.getRight() is entity (http body)
-            return pair.getLeft().post(pair.getRight());
-        });
-        // If check status failed, throw client exception.
-        checkStatus(response, Response.Status.CREATED,
-                    Response.Status.OK, Response.Status.ACCEPTED);
-        return new RestResult(response);
+        Request.Builder requestBuilder = getRequestBuilder(path, null, headers, params);
+        requestBuilder.post(buildRequestBody(object, headers));
+
+        try (Response response = this.request(
+                () -> client.newCall(requestBuilder.build()).execute())) {
```

Comment:
seems we don't need to callback anymore, just `this.request(requestBuilder)` is ok

Reply:
seems this is for the RestClientTest to mock the request action